### PR TITLE
[WIP] Qrexec docs rewrite: internals (Phase 3)

### PR DESCRIPTION
**Pull request by pierwill**
Opening this PR to begin work on the [qrexec internals](https://www.qubes-os.org/doc/qrexec-internals/) page. This is the third and final phase of the qrexec documentation project.

(For previous work see [Phase 1](https://github.com/QubesOS/qubes-doc/pull/847) and [Phase 2](https://github.com/QubesOS/qubes-doc/pull/856).)

### CODE REVIEW

File: developer/services/qrexec.md [hunk 1]
```diff
@@ -66,6 +66,8 @@ Thus, it is usually more convenient to use `qvm-run`.
 There can be an almost arbitrary number of `qrexec-client` processes for a given domain.
 The limiting factor is the number of available vchan channels, which depends on the underlying hypervisor, as well the domain's OS.
 
+For more details on the qrexec framework and protocol, see "[Qubes RPC internals](/doc/#qrexec-internals)."
```
Comment (andrewdavidwong) [inline developer/services/qrexec.md]: Is the `#` in the URL a typo?

File: developer/services/qrexec-internals.md [hunk 2]
```diff
@@ -15,20 +15,26 @@ redirect_from:
...
 * `/usr/lib/qubes/qrexec-daemon`: One instance is required for every active domain. Responsible for:
   * Handling execution and service requests from **dom0** (source: `qrexec-client`).
   * Handling service requests from the associated domain (source: `qrexec-client-vm`, then `qrexec-agent`).
 * Command line: `qrexec-daemon domain-id domain-name [default user]`
 * `domain-id`: Numeric Qubes ID assigned to the associated domain.
 * `domain-name`: Associated domain name.
 * `default user`: Optional. If passed, `qrexec-daemon` uses this user as default for all execution requests that don't specify one.
+
+### qrexec-policy
 * `/usr/lib/qubes/qrexec-policy`: Internal program used to evaluate the RPC policy and deciding whether a RPC call should be allowed.
```
Comment (marmarek) [inline developer/services/qrexec-internals.md]: It's in `/usr/bin/qrexec-policy` now.

File: developer/services/qrexec-internals.md [hunk 3]
```diff
@@ -15,20 +15,26 @@ redirect_from:
...
+
+### qrexec-client
 * `/usr/lib/qubes/qrexec-client`: Used to pass execution and service requests to `qrexec-daemon`. Command line parameters:
```
Comment (marmarek) [inline developer/services/qrexec-internals.md]: This is also moved to `/usr/bin/qrexec-client`, but compatibility symlink remains in `/usr/lib/qubes`.

File: developer/services/qrexec-internals.md [hunk 4]
```diff
@@ -15,20 +15,26 @@ redirect_from:
...
-Components residing in the same domain (`qrexec-client-vm` to `qrexec-agent`, `qrexec-client` to `qrexec-daemon`) use pipes as the underlying transport medium, while components in separate domains (`qrexec-daemon` to `qrexec-agent`, data channel between `qrexec-agent`s) use vchan link.
+The qrexec framework consists of a number of processes communicating with each other using common IPC protocol (described in detail below).
+Components residing in the same domain (`qrexec-client-vm` to `qrexec-agent`, `qrexec-client` to `qrexec-daemon`) use local sockets as the underlying transport medium.
+Components in separate domains (`qrexec-daemon` to `qrexec-agent`, data channel between `qrexec-agent`s) use vchan links.
 Because of [vchan limitation](https://github.com/qubesos/qubes-issues/issues/951), it is not possible to establish qrexec connection back to the source domain.
 
 ## Dom0 tools implementation
 
+### qrexec-daemon
 * `/usr/lib/qubes/qrexec-daemon`: One instance is required for every active domain. Responsible for:
```
Comment (marmarek) [inline developer/services/qrexec-internals.md]: This is now in `/usr/sbin/qrexec-daemon`, but compatibility symlink remains in `/usr/lib/qubes`.

**Comment by pierwill** [comment]
Thanks, Marek! I'll fix these when I can get back around to working on this.

**Comment by pierwill** [comment]
There is more work I'd like to do here, but not sure when. Feel free to merge!

**Pull request merged**